packages/material-ui/src/RadioGroup/RadioGroup.js

```diff
@@ -9,13 +9,14 @@ import { createChainedFunction, find } from '../utils/helpers';
 class RadioGroup extends React.Component {
   radios = [];
 
-  state = {
-    value: null,
-  };
-
   constructor(props) {
     super();
-    this.isControlled = props.value != null;
+
+    this.isControlled = !props.hasOwnProperty('defaultValue') && !props.hasOwnProperty('value');
```

Comment:
I have checked how React handles the uncontrolled/controlled warnings. They look for `value !== undefined`. I don't think that we need to use `hasOwnProperty`.
In v4, I think that we should use `value !== undefined` over `value != null`. https://codesandbox.io/s/5v6vmxnx6n

Reply:
for v3 you think it should remain `value != null` ?

Reply:
I would say yes for consistency with the other components?

Reply:
ok :) done

Reply:
Not sure what this codesandbox should illustrate. React treats any nullish value as uncontrolled. Therefore using `value !== undefined` does not match React. React even warns against `null` for input values. `value` takes precedence over `defaultValue` if not nullish (`undefined | null`).

Playground: https://codesandbox.io/s/7zp797xny1
React source: 
https://github.com/facebook/react/blob/c954efa70f44a44be9c33c60c57f87bea6f40a10/packages/react-dom/src/client/ReactDOMInput.js#L37-L40

Reply:
What about this check: https://github.com/facebook/react/blob/c954efa70f44a44be9c33c60c57f87bea6f40a10/packages/react-dom/src/client/ReactDOMInput.js#L78-L79

Reply:
Ok, you are right, React handles the warning and the input behavior differently.

Reply:
I'm removing the break change suggestion: declined.